### PR TITLE
Added C++ Solution 

### PR DESCRIPTION
**Pull request by Anshul1507**
File contains 
- [x] 31. Next Permutation

### CODE REVIEW

File: cpp/_31.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Solution {
+public:
+    void nextPermutation(vector<int>& nums) {
+    
+        int i1,i2;
```
Comment (fishercoder1534) [inline cpp/_31.cpp:5]: one variable per line please.

File: cpp/_31.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                }
+            }
+            swap(nums[i1],nums[j]);
+            reverse(nums.begin()+i1+1,nums.end());
+        }else{
+            sort(nums.begin(),nums.end());
+        }
+        for(auto i:nums){
+            cout << i << " ";
+        }
```
Comment (fishercoder1534) [inline cpp/_31.cpp:31]: please remove debug statements.

**Review by fishercoder1534 — CHANGES_REQUESTED** [review]

File: cpp/_31.cpp (hunk 2 above)
Comment (Anshul1507) [reply cpp/_31.cpp:31]: these are o/p (function is void).

File: cpp/_31.cpp (hunk 1 above)
Comment (Anshul1507) [reply cpp/_31.cpp:5]: sure

File: cpp/_31.cpp (hunk 2 above)
Comment (Anshul1507) [reply cpp/_31.cpp]: sure

**Comment by Anshul1507** [comment]
Hey @fishercoder1534, changes are done.

**Review by fishercoder1534 — APPROVED** [review]

**Pull request merged**